File: iteration2/CSE3063F21P1_GRP16/src/main/java/Models/E2prerequisite.java
package Models;

import java.util.ArrayList;

public class E2prerequisite  implements CourseEvents {
	
	
private ArrayList<Student> students;


public ArrayList<Student> getStudents() {
	return students;
}

public  E2prerequisite() {
	students=new ArrayList<>();
	
}

@Override
public String toString() {
		
	String s="Total Number of Student who could't take the course becuase of prerequisites: "+students.size()+"\nList Of Students:\n";
	
	for(int i=0;i<students.size();i++) {
		s+=students.get(i).getFirstName()+" "+students.get(i).getLastName()+"\n";
		
		
	}
	return s;
}

}

File: iteration3/CSE3063F21P1_GRP16/models/E2prerequisite.py

from typing import List

from CourseEvents import CourseEvents

from Student import Student


class E2prerequisite(CourseEvents):
    def __init__(self):
        self._students = []

    def toString(self):
        s = "    Total Number of Student who couldn't take the course because of prerequisites: {}\n".format(len(self._students))
        if len(self._students) > 0:
            s += "     -List Of students:\n"
        else:
            s += "\n"

        for i in range(len(self._students)):
            s += "      *{} {}\n".format(self._students[i].getFirstName(), self._students[i].getLastName())

        return s
